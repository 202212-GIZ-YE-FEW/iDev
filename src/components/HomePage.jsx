import { withTranslation } from "next-i18next";

import CommunicateThroughSection from "./Homepage/CommunicateThroughSection";
import HeroSection from "./Homepage/HeroSection";
import RecentBlogSection from "./Homepage/RecentBlogSection";
import TicketSection from "./Homepage/TicketSection";
import TrustTherapistSection from "./Homepage/TrustTherapistSection";
function HomePage() {
    return (
        <>
            <HeroSection />
            <section className='bg-yellow py-16'>
                <TrustTherapistSection />
            </section>
            <section className='bg-white py-20'>
                <CommunicateThroughSection />
            </section>
            <section className='bg-light-cyan py-20 id="recentblog'>
                <RecentBlogSection />
            </section>
            <section className='bg-white py-20'>
                <TicketSection />
            </section>
        </>
    );
}
export default withTranslation("home")(HomePage);
