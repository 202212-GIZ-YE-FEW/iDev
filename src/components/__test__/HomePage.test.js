import { QueryClient, QueryClientProvider } from "react-query";
import renderer from "react-test-renderer";

import HomePage from "../HomePage";

const queryClient = new QueryClient();

it("renders correctly", () => {
    const tree = renderer
        .create(
            <QueryClientProvider client={queryClient}>
                <HomePage />
            </QueryClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
