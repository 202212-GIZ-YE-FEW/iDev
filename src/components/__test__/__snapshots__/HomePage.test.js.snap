// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="min-w-[96vw] bg-cover"
    style={
      Object {
        "backgroundImage": "url(/header.png)",
        "minHeight": "calc(100vh - 5rem)",
      }
    }
  >
    <div
      className="container h-full flex flex-col lg:flex-row gap-y-10 justify-between lg:mt-32"
    >
      <div
        className="flex flex-col mt-10 lg:mt-0 text-center lg:text-start"
      >
        <h1
          className="text-xl sm:text-3xl lg:text-5xl uppercase"
        >
          weAreHereTo
        </h1>
        <p
          className="text-3xl sm:text-5xl lg:text-9xl mb-3 lg:mb-0 uppercase"
        >
          help
        </p>
        <a
          className="w-fit xl:w-full rtl:w-full"
          href="/appointment"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="relative self-center"
          >
            <button
              className="transform hover:bg-yellow transition duration-500  hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  capitalize text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
              disabled={false}
              onClick={[Function]}
            >
              bookAppointment
            </button>
          </div>
        </a>
      </div>
      <div
        className="basis-1/2"
      >
        <img
          alt=""
          className="max-w-full lg:min-w-[20rem] lg:max-w-[27rem] xl:min-w-[40rem]"
          data-aos="zoom-in-up"
          data-nimg="1"
          decoding="async"
          height={200}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fhero.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fhero.png&w=640&q=75 1x, /_next/image?url=%2Fhero.png&w=1080&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={500}
        />
      </div>
    </div>
  </header>,
  <section
    className="bg-yellow py-16"
  >
    <div
      className="container flex flex-col space-y-8 justify-between"
    >
      <div
        className="mb-5 text-start"
      >
        <span
          className="font-normal block text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xl uppercase break-words"
        >
          trustTherapistTitle
        </span>
      </div>
      <img
        alt="trust_therapist"
        data-nimg="1"
        decoding="async"
        height={300}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Ftrust_therapists.png&w=1200&q=75"
        srcSet="/_next/image?url=%2Ftrust_therapists.png&w=640&q=75 1x, /_next/image?url=%2Ftrust_therapists.png&w=1200&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={600}
      />
      <p
        className="pb-20"
      >
        trustTherapistSubtitle
        .
      </p>
      <a
        className="w-fit xl:w-full rtl:w-full"
        href="/appointment"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="relative self-center"
        >
          <button
            className="transform hover:bg-yellow transition duration-500  hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
            disabled={false}
            onClick={[Function]}
          >
            bookAppointment
          </button>
        </div>
      </a>
    </div>
  </section>,
  <section
    className="bg-white py-20"
  >
    <div
      className="container flex flex-col space-y-20 justify-between"
    >
      <div
        className="mb-5 text-start"
      >
        <span
          className="font-normal block text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xl uppercase break-words"
        >
          commThroughTitle
        </span>
      </div>
      <div
        className="grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5"
      >
        <div
          className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
          data-aos="flip-down"
        >
          <img
            alt="voiceCall svg"
            data-nimg="1"
            decoding="async"
            height={50}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/home/voice.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={50}
          />
          <span
            className="uppercase text-4xl"
          >
            voiceCall
          </span>
          <p
            className="text-sm rtl:text-base text-black/50"
          >
            voiceCallDesc
          </p>
        </div>
        <div
          className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
          data-aos="flip-down"
        >
          <img
            alt="chat svg"
            data-nimg="1"
            decoding="async"
            height={50}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/home/chat.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={50}
          />
          <span
            className="uppercase text-4xl"
          >
            chat
          </span>
          <p
            className="text-sm rtl:text-base text-black/50"
          >
            chatDesc
          </p>
        </div>
        <div
          className="bg-light-white flex flex-col items-center justify-center text-center p-9 rounded-3xl space-y-3 drop-shadow-lg"
          data-aos="flip-down"
        >
          <img
            alt="videoCall svg"
            data-nimg="1"
            decoding="async"
            height={50}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/home/video.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={50}
          />
          <span
            className="uppercase text-4xl"
          >
            videoCall
          </span>
          <p
            className="text-sm rtl:text-base text-black/50"
          >
            videoCallDesc
          </p>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="bg-light-cyan py-20 container"
  >
    <div
      className="flex flex-col space-y-16 justify-between"
    >
      <p
        className="text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xlfont-normal block uppercase break-words"
      >
        common:recentBlogsTitle
      </p>
      <div
        className="react-multi-carousel-list  "
        dir="ltr"
      >
        <ul
          className="react-multi-carousel-track "
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "overflow": "unset",
              "transform": "translate3d(0px,0,0)",
              "transition": "none",
            }
          }
        />
      </div>
    </div>
  </section>,
  <section
    className="bg-white py-20"
  >
    <div
      className="container flex flex-col justify-between"
    >
      <div
        className="bg-light-white flex flex-col items-center justify-center text-center py-10 px-5 rounded-md space-y-6 drop-shadow-lg"
      >
        <span
          className="uppercase text-4xl"
        >
          askIfCounselor
        </span>
        <span
          className="text-base rtl:text-xl text-black"
        >
          askCounselor
        </span>
        <div>
          <a
            href="/requirements"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="relative self-center"
            >
              <button
                className="transform hover:bg-yellow transition duration-500  hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
                disabled={false}
                onClick={[Function]}
              >
                learnMore
              </button>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;
